File: abc137_c.py
# https://atcoder.jp/contests/abc137/tasks/abc137_a

from collections import defaultdict
d = defaultdict(int)
N = int(input())
for i in range(N):
    s = input()
    ss = ''.join(sorted(s))
    d[ss] += 1
ans = 0
for k, v in d.items():
    ans += (v * (v-1)) //2
print(ans)


# N = int(input())
# d = dict()
# for i in range(N):
#     s = list(input())
#     s.sort()
#     s = ''.join(s)
#     if s in d:
#         d[s] += 1
#     else:
#         d[s] = 1

# ans = 0
# for i in d.values():
#     ans += (i * (i-1)) // 2

# print(ans)


# N = int(input())
# ss = []
# for i in range(N):
#     s = list(input())
#     s.sort()
#     ss.append(s)

# ss.sort()
# ans = 0
# tmp = 1
# for i in range(N-1):
#     if ss[i] == ss[i+1]:
#         tmp += 1
#         if i == N-2:
#             ans += (tmp * (tmp-1)) // 2
#     else:
#         ans += (tmp * (tmp-1)) // 2
#         tmp = 1

# print(ans)

File: atcoder/abc137_c.cpp
// https://atcoder.jp/contests/abc137/tasks/abc137_c
// https://atcoder.jp/contests/abc137/submissions/6835088
#include<iostream>
#include<algorithm>
//#include <numeric>
//#include<string>
//#include<vector>
#include<map>
//#include<tuple>
//#include<queue>
//#include<regex>

// #include <bits/stdc++.h>
#define rep(i,n) for (int i = 0; i < (n); ++i)
using namespace std;
typedef long long ll;

int main() {
	int n;
	cin >> n;
	map<string, int> mp;
	rep(i, n) {
		string s;
		cin >> s;
		sort(s.begin(), s.end());
		mp[s]++;
	}
	ll ans = 0;
	for (auto& p : mp) {
		int s = p.second;
		ans += (ll)s * (s - 1) / 2;
	}
	cout << ans << endl;
	return 0;
}
